### PR TITLE
feat: huobi ws provider

### PR DESCRIPTION
**Pull request by aljo242**
Closes BLO-736

### CODE REVIEW

File: providers/websockets/huobi/ws_data_handler.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+		subscriptionResponse SubscriptionResponse
+		tickerStream         TickerStream
+	)
+
+	reader, err := gzip.NewReader(bytes.NewReader(message))
+	if err != nil {
+		h.logger.Error("error creating gzip reader", zap.Error(err))
+		return resp, nil, err
+	}
+	defer reader.Close()
```
Comment (davidterpay) [inline providers/websockets/huobi/ws_data_handler.go:78]: does the defer need to happen before the error?

File: providers/websockets/huobi/ws_data_handler.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+		h.logger.Error("error reading gzip", zap.Error(err))
+		return resp, nil, err
+	}
+
+	// attempt to unmarshal to ping and check if field values are not nil
+	if err := json.Unmarshal(uncompressed.Bytes(), &pingMessage); err == nil && pingMessage.Ping != 0 {
+		h.logger.Debug("received ping message")
+		updateMessage, err := NewPongMessage(pingMessage)
+		return resp, updateMessage, err
+	}
```
Comment (davidterpay) [inline providers/websockets/huobi/ws_data_handler.go:92]: should we add a test case for this?

**Review by davidterpay — COMMENTED** [review]
lgtm, would add the test case for pongs

File: providers/websockets/huobi/ws_data_handler.go (hunk 1 above)
Comment (aljo242) [reply providers/websockets/huobi/ws_data_handler.go:78]: If there is an error, the reader that will be returned is `nil`

**Review by davidterpay — APPROVED** [review]

**Pull request merged**